components/Actions.js

```diff
@@ -1,59 +1,68 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { View, StyleSheet } from 'react-native'
 import { IconButton } from 'react-native-paper'
 
 import theme from '../lib/defaultTheme'
 
+import IconButtonWithLabel from './IconButtonWithLabel'
+
 const A11Y_LABELS = {
-  scan: 'Add new secret by scanning a QR Code',
-  upload: 'Add new secret by uploading a QR Code',
-  type: 'Add new secret by typing the details',
+  scan: 'Scan QR Code',
+  upload: 'Upload',
+  type: 'Add details manually',
 }
 
 export default function Actions({
   onScanNewSecretScreen,
   onUploadNewSecretScreen,
   onTypeNewSecretScreen,
 }) {
+  const [actionsVisible, setActionsVisible] = useState(false)
   return (
-    <View style={styles.actions}>
-      <IconButton
-        icon="qrcode"
-        accessibilityLabel={A11Y_LABELS.scan}
-        mode="contained"
-        style={styles.button}
-        size={40}
-        color={theme.colors.surface}
-        onPress={onScanNewSecretScreen}
-      />
+    <View style={styles.container}>
+      {actionsVisible && (
+        <View style={styles.actions}>
+          <IconButtonWithLabel
+            label={A11Y_LABELS.scan}
+            icon="qrcode"
+            onPress={onScanNewSecretScreen}
+          />
+          <IconButtonWithLabel
+            label={A11Y_LABELS.upload}
+            onPress={onUploadNewSecretScreen}
+            icon="upload"
+          />
+          <IconButtonWithLabel
+            label={A11Y_LABELS.type}
+            onPress={onTypeNewSecretScreen}
+            icon="import"
+          />
+        </View>
+      )}
```

Comment:
it would be nice to include an animation at some point. if you prefer not to do it here, can you please create an issue to track it?

Reply:
Yes, good shout. I'll create a ticket to add micro-interactions.